common/src/db.rs

```diff
@@ -0,0 +1,261 @@
+use anyhow::Context;
+use migration::{async_trait::async_trait, Migrator, MigratorTrait};
+use postgresql_embedded::PostgreSQL;
+use sea_orm::{
+    prelude::async_trait, ConnectOptions, ConnectionTrait, DatabaseConnection, DatabaseTransaction,
+    DbBackend, DbErr, ExecResult, QueryResult, Statement,
+};
+use std::ops::{Deref, DerefMut};
+use std::sync::Arc;
+use tempfile::TempDir;
+
+#[derive(Copy, Clone)]
+pub enum Transactional<'db> {
+    None,
+    Some(&'db DatabaseTransaction),
+}
+
+impl<'db> From<&'db DatabaseTransaction> for Transactional<'db> {
+    fn from(inner: &'db DatabaseTransaction) -> Self {
+        Self::Some(inner)
+    }
+}
+
+#[derive(Clone)]
+pub enum ConnectionOrTransaction<'db> {
+    Connection(&'db DatabaseConnection),
+    Transaction(&'db DatabaseTransaction),
+}
+
+#[async_trait::async_trait]
+impl ConnectionTrait for ConnectionOrTransaction<'_> {
```

Comment:
Why do we need to reimplement [upstream](https://docs.rs/sea-orm/latest/sea_orm/trait.ConnectionTrait.html)?

Reply:
That was there before, I just moved it. Not sure if we need it, I can try to remove it.

Reply:
It was due to an initial foray into allowing transactional or non-transactional usage, and was probably mis-guided.

I've been back towards touching txns now, still trying to sort out better patterns.